File: packages/matlab_tests/p_extract_geom.m
% mp-declare-processor matlab_tests.extract_geom
% mp-declare-version 0.1
% mp-declare-description Extract a subset of channels from a geom.csv file
% mp-declare-input geom
% mp-declare-output geom_out
% mp-declare-parameter channels
function p_extract_geom(geom,geom_out,params)
G=csvread(geom);
chans=str2num(params.channels);
G=G(chans,:);
csvwrite(geom_out,G);

File: packages/mlpy/p_extract_geom.py
#!/usr/bin/python3

import numpy as np

class Processor:
	name='mlpy.extract_geom'
	description='Extract a subset of channels from a geom.csv file'
	version='0.1'
	inputs=[{"name":"geom","description":""}]
	outputs=[{"name":"geom_out","description":""}]
	parameters=[{"name":"channels","description":"","optional":False}]
	def run(self,args):
		geom_path=args['geom']
		geom_out_path=args['geom_out']
		channels_str=args['channels']
		channels=np.fromstring(channels_str,dtype=int,sep=',')
		X=np.loadtxt(open(geom_path, "rb"), delimiter=",")
		X=X[channels-1,:]
		np.savetxt(geom_out_path,X,delimiter=",",fmt="%g")
		return True
